src/lib/support/ZclString.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ZclString.h"
+
+namespace chip {
+
+// maximum size of MutableByteSpan -1 (the length of pascal string)
```

Comment:
the comment does not sem to really explain things: why the reference to MutableByteSpan? That one has `    size_t mDataLen;` so its size can be quite large.

I would understand a comment of:

```
// ZCL strings are stored as pascal-strings (first byte contains the length of the
// data), so the maximum string length is the maximum of uint8_t
```

however I do not understand the -1. Please explain ideally referencing some spec.